smart_open/s3.py

```diff
@@ -232,12 +232,10 @@ def readline(self, limit=-1):
             #
             if self._line_terminator in self._buffer:
                 next_newline = self._buffer.index(self._line_terminator)
```

Comment:
Unrelated to this PR, but this seems inefficient: why search for `_line_terminator` twice? Here and in the line above? This is a linear, potentially costly operation, so we should only do it once.

Reply:
You're right about it being costly, but when I wrote this code, I measured the alternative (try-catch), and that was slower (see the comment above this function). This behavior depends on the data you're processing, and what proportion of reads end up having the line terminator inside the buffer.